Tests/SonarScanner.MSBuild.PreProcessor.Test/PreprocessorObjectFactoryTests.cs

```diff
@@ -84,16 +84,18 @@ public void CreateHttpClient_Authorization()
             AuthorizationHeader("admin", "password").Should().Be("Basic YWRtaW46cGFzc3dvcmQ=");
 
             static string AuthorizationHeader(string userName, string password) =>
-                GetHeader(PreprocessorObjectFactory.CreateHttpClient(userName, password, null, null), HttpRequestHeader.Authorization);
+                GetHeader(PreprocessorObjectFactory.CreateHttpClient(userName, password, null, null), "Authorization");
         }
 
         [TestMethod]
         public void CreateHttpClient_UserAgent()
         {
-            var userAgent = GetHeader(PreprocessorObjectFactory.CreateHttpClient(null, null, null, null), HttpRequestHeader.UserAgent);
-
             var scannerVersion = typeof(WebClientDownloaderTest).Assembly.GetName().Version.ToDisplayString();
-            userAgent.Should().Be($"ScannerMSBuild/{scannerVersion}");
+            var client = PreprocessorObjectFactory.CreateHttpClient(null, null, null, null);
+            GetHeader(client, "User-Agent").Should().Be($"SonarScanner-for-.NET/{scannerVersion}");
+
+            // This asserts wrong "UserAgent" header. Should be removed as part of https://github.com/SonarSource/sonar-scanner-msbuild/issues/1421
+            GetHeader(client, "UserAgent").Should().Be($"ScannerMSBuild/{scannerVersion}");
```

Comment:
For a while we will return different products, it's gonna be confusing. 

On the other hand, it is more convenient for the users that check this, to have to do the change only once.